src/components/WeekDayPicker/__test__/weekDayPicker.spec.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { useUniqueIdentifier } from '../../../libs/hooks';
+import WeekDayPicker from '../';
+import StyledTextError from '../../Input/styled/errorText';
+
+jest.mock('../../../libs/hooks/useUniqueIdentifier', () => jest.fn(() => 'week-day'));
+
+describe('<WeekDayPicker />', () => {
+    it('should render a label when label prop is passed', () => {
+        const component = mount(<WeekDayPicker label="My label" />);
+        expect(component.find('legend').length).toBe(1);
+    });
+    it('should render bottom help text when bottomHelpText prop is passed', () => {
+        const component = mount(<WeekDayPicker bottomHelpText="My help text" />);
+        expect(component.find('div[children="My help text"]').length).toBe(1);
+    });
+    it('should render required asterisk when required prop is true', () => {
+        const component = mount(<WeekDayPicker label="My label" required />);
+        expect(component.find('RequiredAsterisk').exists()).toBe(true);
+    });
+    it('should have input type as checkbox when multiple is true', () => {
+        const component = mount(<WeekDayPicker multiple />);
+        expect(
+            component
+                .find('input')
+                .at(0)
+                .prop('type'),
+        ).toBe('checkbox');
+    });
+    it('should have input type as radio when multiple is false', () => {
+        const component = mount(<WeekDayPicker />);
+        expect(
+            component
+                .find('input')
+                .at(0)
+                .prop('type'),
+        ).toBe('radio');
+    });
+    it('should render an error text when error prop is passed', () => {
+        const component = mount(<WeekDayPicker error="error" />);
+        expect(component.find(StyledTextError).exists()).toBe(true);
+    });
+    it('should have the opposite days disabled when a list of availableDates is sent', () => {
+        const component = mount(<WeekDayPicker availableDates={['sunday', 'monday']} />);
+        expect(
+            component
+                .find('input')
+                .at(5)
+                .prop('disabled'),
+        ).toBe(true);
```

Comment:
try to find a way to iterate through the array of day and check who of them are disabled and who not